Added session handling for JSESSIONID cookie returned by preauthentication token process

tturnerdev opened the pull request:
This is to support pre-authentication token handling. Ref: http://community.jaspersoft.com/documentation/jasperreports-authentication-cookbook/v56/token-based-authentication

tturnerdev committed:
Added session handling for JSESSIONID cookie returned by preauthentication token process

ronindesign commented:
Any better / more intuitive implementations are welcome. Any comments / suggestions?

tturnerdev committed:
Added persistence to initial url target after cookie token auth

tturnerdev committed:
Added possible simplification for JRS token auth

Would replace JRS JSESSIONID cookie exchange, appending auth request token to every request.
